app/controllers/consent_responses_controller.rb

```diff
@@ -83,7 +83,24 @@ def update
   def update_gillick
     @patient_session.assign_attributes(patient_session_gillick_params)
     if @patient_session.save(context: :edit_gillick)
-      redirect_to action: :edit_consent
+      if @patient_session.gillick_competent?
+        redirect_to action: :edit_consent
+      else
+        ActiveRecord::Base.transaction do
+          @draft_consent_response.update!(
+            recorded_at: Time.zone.now,
+            consent: "not_provided"
+          )
+          @patient_session.do_gillick_assessment!
```

Comment:
It would be nice if we continued the pattern of calling this event regardless of the outcome of gillick competence test, so that when successful the patient session transitions to an appropriate state.